managed/CounterStrikeSharp.API/Core/GlobalContext.cs

```diff
@@ -93,8 +69,38 @@ public void LoadAll()
                 });
             }
 
-            Console.WriteLine("Loading .NET modules...");
+            Console.WriteLine("Loading C# plugins...");
+            int pluginCount = LoadAllPlugins();
+            Console.WriteLine($"All managed modules were loaded. {pluginCount} plugins loaded.");
 
+            RegisterPluginCommands();
+        }
+
+        public void LoadPlugin(string path)
+        {
+            var plugin = new PluginContext(path);
+
+            try
+            {
+                foreach (var existingPlugin in _loadedPlugins)
+                {
+                    if (plugin.Name == existingPlugin.Name)
+                    {
+                        throw new Exception("Plugin is already loaded.");
+                    }
+                }
+                _loadedPlugins.Add(plugin);
+                plugin.Load();
+            }
+            catch (Exception e)
+            {
+                _loadedPlugins.Remove(plugin);
+                throw;
+            }
```

Comment:
I think if we just add the plugin to the loaded list _after_ we load, then we can ignore the whole exception delete flow.
```suggestion
            foreach (var existingPlugin in _loadedPlugins)
            {
                if (plugin.Name == existingPlugin.Name)
                {
                    throw new Exception("Plugin is already loaded.");
                }
            }
            plugin.Load();
            _loadedPlugins.Add(plugin);
```